File: Solutions/91/src.cpp
class Solution {
public:
    int numDecodings(string s) {
        int *dp = new int[s.size() + 1]();
        dp[0] = 1; 
        dp[1] = ( (s[0] == '0') ? 0 : 1);
        for(int i = 2; i <= s.size(); i++)
        {
            if(s[i-1] == '0' && !(s[i-2] == '1' || s[i-2] == '2') )
                return 0;
            if( (s[i-1] >= '1' && s[i-1] <= '6' && s[i-2] == '2' )
               || (s[i-1] >= '1' && s[i-1] <= '9' && s[i-2] == '1') )
                dp[i] = dp[i-1] + dp[i-2];
            else if(s[i-1] == '0')
                dp[i] = dp[i-2];
            else
                dp[i] = dp[i-1];
        }
        return dp[s.size()];
    }
};

File: Solutions/1003/src.py
class Solution:
    def isValid(self, S: str) -> bool:
        stack = []
        for c in S:
            if len(stack) >= 2 and c == 'c' and stack[len(stack) - 1] == 'b' and stack[len(stack) -2] == 'a':
                stack.pop()
                stack.pop()
            else:
                stack.append(c)
        if len(stack) != 0:
            return False
        else:
            return True